enterprise/views.py

```diff
@@ -231,13 +244,15 @@ def post_course_specific_consent(self, request, course_id, consent_provided):
         if not request.user.is_authenticated():
             raise Http404
 
-        EnterpriseCourseEnrollment.objects.update_or_create(
-            enterprise_customer_user__user_id=request.user.id,
-            course_id=course_id,
-            defaults={
-                'consent_granted': consent_provided,
-            }
-        )
+        enrollment_deferred = request.POST.get('enrollment_deferred')
+        if enrollment_deferred is None:
+            EnterpriseCourseEnrollment.objects.update_or_create(
+                enterprise_customer_user__user_id=request.user.id,
+                course_id=course_id,
+                defaults={
+                    'consent_granted': consent_provided,
+                }
+            )
```

Comment:
Hmm, I can't think of a way to separate this out that would make it easier to understand. It's one condition gating one statement that needs to stay in the same execution order relative to the surrounding checks. So if I factor it out, most of the code would be repeated, and then if I upstreamed the common code to `post()`, the functions would vanish and everything would just be inlined.
Do you think it would suffice to add a comment explaining `enrollment_deferred` instead?